src/a-runtime-console/kubernetes/ui/pipeline/input-action-dialog/input-action-dialog.component.ts

```diff
@@ -52,31 +48,29 @@ export class InputActionDialog {
       if (url.startsWith('//')) {
         url = url.substring(1);
       }
+      let jenkinsUrl = this.build.jenkinsBuildURL;
+      // Remove /job from jenkinsBuildURL if exists
+      let idx = jenkinsUrl.indexOf("/job/");
+      if (idx > -1) {
+        jenkinsUrl = jenkinsUrl.substring(0, idx);
+      }
       // lets replace URL which doesn't seem to work right ;)
       const postfix = '/wfapi/inputSubmit?inputId=Proceed';
       if (url.endsWith(postfix)) {
         url = url.substring(0, url.length - postfix.length) + '/input/Proceed/proceedEmpty';
       }
-
-      let jenkinsNamespace = this.build.jenkinsNamespace;
-      let forgeUrl = this.forgeApiUrl;
-      if (!forgeUrl) {
-        console.log('Warning no $FABRIC8_FORGE_API_URL environment variable!');
-      } else if (!jenkinsNamespace) {
-        console.log('Warning no jenkinsNamespace on the Build!');
-      } else {
-        url = pathJoin(forgeUrl, '/api/openshift/services/jenkins/', jenkinsNamespace, url);
-        let token = this.authService.getToken();
-        console.log('about to invoke ' + url);
+      url = pathJoin(jenkinsUrl, url);
+      this.authService.getOpenShiftToken().subscribe(token => {
         let options = new RequestOptions();
         let headers = new Headers();
         headers.set('Authorization', 'Bearer ' + token);
+        headers.set('Content-Type', 'text/plain');
         options.headers = headers;
         let body = null;
         this.http.post(url, body, options).subscribe(res => {
```

Comment:
@gastaldi could you please shed some light into how this works? I am asking to ensure that fixing https://github.com/fabric8-services/fabric8-wit/issues/2095 solves the issue for UI.

Reply:
@sthaha this is an Ajax code that makes a POST request to the URL containing the Jenkins URL. It should have the same behavior as if you have clicked the Promote button inside Jenkins